trunner/test_runner.py

```diff
@@ -244,6 +244,10 @@ def run_tests(self, tests: Sequence[TestOptions]) -> Sequence[TestResult]:
         # Ensure first test will start with reboot
         last_test_failed = True
 
+        # Unload data before starting tests
+        if self.ctx.should_flash is not False and self.ctx.target.experimental is False:
```

Comment:
I don't understand why it's disabled for experimental targets.

Can't we just always drain the buffer (without any IF?) - the same way as below:
`self.target.dut.read(timeout=0.1)`

Please note that this is still a race - new data might be produced between reading the buffer and issuing reboot before the first test.